### PR TITLE
[libzip] Remove building of i386 on MacOS.

### PR DESCRIPTION
**Pull request by dellis1972**
Xcode 10.0 deprecates compiling of 32bit applications.

	The macOS 10.14 SDK no longer contains support for compiling
	32-bit applications. If developers need to compile for i386, Xcode
	9.4 or earlier is required. (39858111)

So just build 64bit everywhere.

### CODE REVIEW

**Comment by grendello** [comment]
Should we add a note in the building instructions about this?

**Comment by jonpryor** [comment]
[It doesn't build](https://jenkins.mono-project.com/job/xamarin-android-pr-builder/4120/):

```
xamarin-android/src/libzip/libzip.csproj(25,3): error MSB4024: The imported project file "/Users/builder/jenkins/workspace/xamarin-android-pr-builder/xamarin-android/src/libzip/libzip.p
rojitems" could not be loaded. An error occurred while parsing EntityName. Line 6, position 101.
```

File: src/libzip/libzip.projitems [hunk 1]
```diff
@@ -3,7 +3,7 @@
   <ItemGroup>
     <_LibZipTarget Include="host-Darwin" Condition="$(AndroidSupportedHostJitAbisForConditionalChecks.Contains (':Darwin:'))">
       <CMake>cmake</CMake>
-      <CMakeFlags>-C$(MSBuildThisFileDirectory)/CMakeCacheInitialOSX.txt -DCMAKE_OSX_ARCHITECTURES=&quot;i386;x86_64&quot; -DBUILD_SHARED_LIBS=ON</CMakeFlags>
+      <CMakeFlags>-C$(MSBuildThisFileDirectory)/CMakeCacheInitialOSX.txt -DCMAKE_OSX_ARCHITECTURES=&quotx86_64&quot; -DBUILD_SHARED_LIBS=ON</CMakeFlags>
```
Comment (jonpryor) [inline src/libzip/libzip.projitems]: The problem is you need `&quot;` not `&quot` (note missing semicolon).

**Comment by jonpryor** [comment]
I'd also like to use the ~same commit message as Java.Interop: https://github.com/xamarin/java.interop/commit/94c6bb570b67cdd9634f0e487a348cd75c600541

```
[Xcode 10.0 deprecates compiling of 32-bit applications][0].

> The macOS 10.14 SDK no longer contains support for compiling
> 32-bit applications.  If developers need to compile for i386,
> Xcode 9.4 or earlier is required. (39858111)

So just build 64-bit everywhere.

[0]: https://developer.apple.com/documentation/xcode_release_notes/xcode_10_release_notes#3035632
```

**Comment by jonpryor** [comment]
This PR needs one more change:

```diff
diff --git a/build-tools/create-bundle/bundle-path.targets b/build-tools/create-bundle/bundle-path.targets
index f9bc3986..1073e9ea 100644
--- a/build-tools/create-bundle/bundle-path.targets
+++ b/build-tools/create-bundle/bundle-path.targets
@@ -25,6 +25,7 @@
     <ItemGroup>
       <VersionFile Include="..\..\src\mono-runtimes\mono-runtimes.*" />
       <VersionFile Include="..\..\src\mono-runtimes\ProfileAssemblies.projitems" />
+      <VersionFile Include="..\..\src\libzip\libzip.*" />
       <VersionFile Include="..\scripts\BuildEverything.mk" />
     </ItemGroup>
     <HashFileContents
```

Without the above change, the existing mono bundle isn't invalidated, and thus [`libzip.csproj` isn't rebuilt](https://jenkins.mono-project.com/view/Xamarin.Android/job/xamarin-android-pr-builder/4124/):

```
Project "/Users/builder/jenkins/workspace/xamarin-android-pr-builder/xamarin-android/build-tools/Xamarin.Android.Tools.BootstrapTasks/Xamarin.Android.Tools.BootstrapTasks.csproj" (7:2) is building "/Users/builder/jenkins/workspace/xamarin-android-pr-builder/xamarin-android/src/libzip/libzip.csproj" (9:2) on node 1 (default targets).
PrepareForBuild:
  Creating directory "obj/Debug/".
_SetCMakeListsTxtTimeToLastCommitTimestamp:
  /usr/local/bin/git log -1 --format=%cd --date=format-local:"%Y/%m/%d %H:%M:%S" 
  Touching "/Users/builder/jenkins/workspace/xamarin-android-pr-builder/xamarin-android/external/libzip/CMakeLists.txt".
_BuildUnlessCached:
Skipping target "_BuildUnlessCached" because all output files are up-to-date with respect to the input files.
Done Building Project "/Users/builder/jenkins/workspace/xamarin-android-pr-builder/xamarin-android/src/libzip/libzip.csproj" (default targets).
```

**Comment by jonpryor** [comment]
Ignoring the [three reported unit test failures](https://jenkins.mono-project.com/job/xamarin-android-pr-builder/4124/testReport/) because those failures are in the BCL, and *our* `libzip` changes should have no impact on `ConditionalWeakTable` and Remoting.

**Pull request merged**